hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/x509/certificate/client/TestDefaultCertificateClient.java

```diff
@@ -532,4 +544,53 @@ public void testRenewAndStoreKeyAndCertificate() throws Exception {
     // a success renew after auto cleanup new key and cert dir
     dnCertClient.renewAndStoreKeyAndCertificate(true);
   }
+
+  @Test
+  public void testCloseCertificateClient(@TempDir File metaDir)
```

Comment:
Other than calling close at the end what is the intention for this new test?

Reply:
We have a certificate lifetime monitor, that is scheduled to be run by a scheduled executor service.
The intention here is to check if after we close the client (which should shut down the executorservice), there are no monitoring thread scheduled to be run. As most of the API is private, the test validates the effects that are visible outside.

Once the client is instantiated, there has to be a scheduled thread with the name compName + "-CertificateLifetimeMonitor", then when the client is closed, and the thread is terminated by the shutdown call in the scheduled executor, the test checks that the thread exists after instantiation, and then it checks there is no thread anymore with the name we expect for the monitor thread after close.

Reply:
Can you please add the same to the test Java doc?

Reply:
yes, done.
I also notified two other places where we potentially do not close a client, I have added that as well.